### DISCUSSION TITLE
Duda sobre cotlang $VALUE#¿?

### DISCUSSION THREAD

**Discussion by gmoplaza** [Cotalker/documentation#219] [2. Questions and Answers] [answered]
#Team tengo dudas sobre cotlang
hice un mini formulario que uso para crear una tarea en una maquina de estado, leyendo la documentación
https://doc.cotalker.com/docs/documentation/admin/workflows/admin_workflow_required_survey/
y haciendo una rutina siguiendo los ejemplos.
Todo bien, pero me saltan algunas dudas, luego de haber leído https://doc.cotalker.com/docs/documentation/automation/admin_cotlang/
en los ejemplos se usan
$VALUE#answer...
$VALUE#meta...
No logro entender exactamente que es lo que puedo poner $VALUE#QUEVALORESVANACÁ. En la documentación le llaman "EXTRACTOR" con la siguiente explicación: "Token values: (1) Values: Dive into value or (2) Operators: Apply function to current data".
Me imaginé que answer era realmente el answer y mirando el modelo pude entender de donde se sacaba la información, pero no logré entender que era meta. De todas formas, más importante es que valores puedo realmente poner ahí y pk? Depende de la acción (en este caso enviar un formulario)?

**Comment by agallardol** [comment, marked as answer]
@gmoplaza En $VALUE tendrás siempre un objeto que representa el "contexto" en que se está ejecutando tu automatización y de este objeto depende qué puedes y qué no escribir como extractor de $VALUE.

Para saber cuál es el contexto de tu automatización, puedes revisar la sección [Triggers & Contexts ](https://doc.cotalker.com/docs/documentation/automation/triggers_and_contexts)

Por ejemplo, si tu automatización se ejecuta cuando alguien envía un slash command en algún channel, entonces el contexto según el link anterior es: 

```
{
  channel: COTChannel,
  message: COTMessage,
  cmdArgs: Array[string]
}
```

y por tanto podrás usar extractors como $VALUE#channel|_id
